src/App.js

```diff
@@ -2,15 +2,64 @@ import React from 'react';
 import './App.scss';
 
 class App extends React.Component {
-  addOne = () => {};
+  state = {
+    number: 0,
+  }
+
+  addOne = () => {
+    this.setState(prevState => ({ number: prevState.number + 1 }));
+  };
 
-  add100 = () => {};
+  add100 = () => {
+    this.setState(prevState => ({ number: prevState.number + 100 }));
+  };
 
-  increase = () => {};
+  increase = () => {
+    this.setState((prevState) => {
+      this.addOne();
+      if (prevState.number > 0 && prevState.number % 5 === 0) {
+        this.add100();
+      }
+    });
+  };
 
```

Comment:
```suggestion
  increase = () => {
    this.addOne();
    if (prevState.number > 0 && prevState.number % 5 === 0) {
      this.add100();
    }
  };
```
No need in setState here. You call two functions that already call setState method inside them